src/functions.php

```diff
@@ -88,6 +88,9 @@ function cast($value)
 
         case "object":
             if (!\method_exists($value, "__toString")) {
+                if ($value instanceof ByteA) {
```

Comment:
This shouldn't be here, as a `ByteA` instance should either have already been converted to a string or this should error.